client/state/action-types.js

```diff
@@ -597,6 +597,8 @@ export const PUSH_NOTIFICATIONS_TOGGLE_ENABLED = 'PUSH_NOTIFICATIONS_TOGGLE_ENAB
 export const PUSH_NOTIFICATIONS_TOGGLE_UNBLOCK_INSTRUCTIONS =
 	'PUSH_NOTIFICATIONS_TOGGLE_UNBLOCK_INSTRUCTIONS';
 export const PUSH_NOTIFICATIONS_UNREGISTER_DEVICE = 'PUSH_NOTIFICATIONS_UNREGISTER_DEVICE';
+export const READER_CONVERSATION_FOLLOW = 'READER_CONVERSATION_FOLLOW';
+export const READER_CONVERSATION_UNFOLLOW = 'READER_CONVERSATION_UNFOLLOW';
```

Comment:
maybe make this MUTE to match the terminology elsewhere

Reply:
Ah - it's not mute :) More like 'revert follow'. I'm using it to undo a follow when an API request fails.

Reply:
Since really what you are looking for isn't unfollow as much as a way to mark a failed follow, it could be nice to reuse `READER_CONVERSATION_FOLLOW` but give the action an error.

Something a-la:
```js
{
  type: READER_CONVERSATION_FOLLOW,
  payload: { postId, siteId }
  error: {
    prevState: FOLLOW | MUTE | null // this is the withRevert part
  }
}
```

Reply:
I'm concerned about reusing `READER_CONVERSATION_FOLLOW` for the error state given we have a lot of `_FAILURE` action types specificially for when things to wrong. Which is the right way and do we need to document that?

It also feels right to have the `prevState` live in the `meta` key to provide some consistency with our other data layer bits, like `post-email-subscriptions/update`.

Reply:
I've had another go at this. I have used `READER_CONVERSATION_UPDATE_FOLLOW_STATUS` to rollback on error (can be used to rollback a failed mute, too). I've also used `meta.prevState` to be consistent with the revert functionality in `post-email-subscriptions/update`. Hope that all sounds reasonable 🤞